lib/ansible/plugins/inventory/foreman.py

```diff
@@ -69,6 +69,13 @@
         type: boolean
         default: False
         version_added: '2.10'
+      legacy_hostvars:
+        description:
+            - Toggle, if true the plugin will build legacy hostvars present in the foreman script
+            - Places hostvars in a dictionary with keys `foreman`, `foreman_facts`, and `foreman_params`
+        type: boolean
+        default: False
+        version_added: '2.10'
```

Comment:
maybe you could add the deprecated flag to this data?

Reply:
You mean, we should basically say in advance something like, "in 2.13, we'll drop support for this (and officially break with the old script)" - is that what mean?

@wenottingham - do we have a general policy about when awx can officially break ties with the old inv. scripts? Guess the answer to that may vary for each inventory source.

Reply:
why would it be deprecated? is it that complex to maintain it as an option?

It's a breaking change regardless of when it happens.

Reply:
@AlanCoding per @wenottingham's note, doesn't sound like we would want to pull this out any time soon.

Also per @wenottingham's note, it seems like this option _is_ something that we'll need to maintain for a while unless (1) there's some other way we plan on providing compatibility in the future (I can't think of any) or (2) we plan on actually transitioning the awx foreman plugin towards its vanilla behavior (but that would be a breaking change and I'm not sure we plan on ever doing that). In short, doesn't seem like we _are_ planning on deprecating support for the new `legacy_hostvars` option.